NEWS.md

```diff
@@ -4,6 +4,9 @@ User-visible changes worth mentioning.
 
 ---
 
+- [#722] `doorkeeper_forbidden_render_options` now supports returning a 404 by
+  specifying `not_found_for_forbidden: true`.
```

Comment:
> Let's rename `not_found_for_forbidden` to `respond_not_found_when_forbidden`?

Done.

> Also I'd add context

Also done. NEWS.md is a little terse, so I wasn't sure how verbose to be.